### ISSUE TITLE
API requests fail unexpectedly

### ISSUE THREAD

**Issue by sirreal** [Automattic/wp-desktop#494]
Using the desktop app in development mode (other setups untested) some requests fail. I believe this is a two-factor issue.

When attempting to test https://github.com/Automattic/wp-calypso/pull/25984 in the desktop app, I ran into problems. Requests begin to fail with the following error:

```json
{"error":"reauthorization_required","message":"A fresh access token must be used to query information about the current user."}
```

This is immediately following a fresh log in. When using Calypso in the browser, I believe this is generally resolved by restoring the two-factor authorization cookie. This is surprising because a two-factor log-in has just been completed. A two-factor re-auth can be triggered by navigating to `/me` (click user icon on top bar). Indeed, the two-factor auth modal is presented in the desktop app, however requests continue to fail in the same way.

These particular failures can be seen by checking out Calypso `update/enable-checklist-all-envs` branch and visiting `/checklist/SITE_SLUG` for a WordPress.com site in the desktop app.

**Comment by sirreal** [comment]
It appears any requests requiring 2fa re-auth (most anything in the `/me` section) fail to load.

**Comment by nsakaimbo** [comment]
_Update_: Another user (@flootr)  just reported this issue when testing the Desktop app. In their profile, they were unable to view their user details under `Account Settings` (see screenshot):

<p align="center">
<img width="350" alt="no-user-details" src="https://user-images.githubusercontent.com/8979548/98025500-beba8100-1dd7-11eb-9363-cbcc549fff09.png">
</p>

- They were unable to see their user details under `/me` or `/me/account` (see screenshot)
- The user was also able to replicate this behavior on Calypso web when using the `SameSite by default cookies` setting (accessed via `chrome://flags`
- Logging out and then back into the application resolved the problem

<p align="center">
<img width="600" alt="chrome-samesite-settings" src="https://user-images.githubusercontent.com/8979548/98025855-3dafb980-1dd8-11eb-849d-bc7fbcfca6eb.png">
</p>

I tried settings this Chrome flag in Electron ([link](https://github.com/Automattic/wp-calypso/compare/debug-settings-authentication)) but so far haven't been able to reproduce this issue.

**Comment by nsakaimbo** [comment]
Update on this thread: we're closing this issue as we expect this and a whole host of other issues to be resolved by the new version of the application (v7.0.5) that was just released. The latest build can now be downloaded at the WordPress Desktop landing page [here](P7jrAc-3-p2).
